packages/datadog-plugin-undici/test/index.spec.js

```diff
@@ -0,0 +1,527 @@
+'use strict'
+
+const getPort = require('get-port')
+const agent = require('../../dd-trace/test/plugins/agent')
+const tags = require('../../../ext/tags')
+const { expect } = require('chai')
+const { rawExpectedSchema } = require('./naming')
+const { DD_MAJOR } = require('../../../version')
+const { ERROR_MESSAGE, ERROR_TYPE, ERROR_STACK } = require('../../dd-trace/src/constants')
+
+const HTTP_REQUEST_HEADERS = tags.HTTP_REQUEST_HEADERS
+const HTTP_RESPONSE_HEADERS = tags.HTTP_RESPONSE_HEADERS
+
+const SERVICE_NAME = DD_MAJOR < 3 ? 'test-http-client' : 'test'
+
+describe('Plugin', () => {
+  let express
+  let fetch
+  let appListener
+
+  describe('undici-fetch', () => {
+    withVersions('undici', 'undici', version => {
+      function server (app, port, listener) {
+        const server = require('http').createServer(app)
+        server.listen(port, 'localhost', listener)
+        return server
+      }
+
+      beforeEach(() => {
+        appListener = null
+      })
+
+      afterEach(() => {
+        if (appListener) {
+          appListener.close()
+        }
+        return agent.close({ ritmReset: false })
+      })
+
+      describe('without configuration', () => {
+        beforeEach(() => {
+          return agent.load('undici', {
+            service: 'test'
+          })
+            .then(() => {
+              express = require('express')
+              fetch = require(`../../../versions/undici@${version}`, {}).get()
+            })
+        })
+
+        afterEach(() => {
+          express = null
+        })
+
+        withNamingSchema(
+          () => {
+            const app = express()
+            app.get('/user', (req, res) => {
+              res.status(200).send()
+            })
+
+            getPort().then(port => {
+              appListener = server(app, port, () => {
+                fetch.fetch(`http://localhost:${port}/user`, { method: 'GET' })
+              })
+            })
+          },
+          rawExpectedSchema.client
+        )
+
+        it('FIRST TEST FAILS BECAUSE.', () => {
+        })
```

Comment:
Leftover from test debugging? 🤔

Reply:
@Qard actually, it's not. There is an issue where whatever _test_ that is first will always fail, hence this empty test. I am hoping that through review we could find a reason why/ or a more acceptable workaround.

Reply:
Strange. 🤔 

What's the error?

Reply:
@Qard no error just eventually times out with 
`Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/runner/work/dd-trace-js/dd-trace-js/packages/datadog-plugin-undici/test/index.spec.js)`